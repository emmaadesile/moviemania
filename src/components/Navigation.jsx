import React from 'react';
import styled from 'styled-components'
import './moviemania-logo.svg';

const Nav = styled.nav`
  height: 92px;
  background: #22262A;
`;

const UpperNav = styled.div`
  height: 52px;
  border: 1px solid #393939;
  display: grid;
  grid-template-columns: 1fr;
  place-items: center;
`;

const LowerNav = styled.div`
  height: 40px;
  display: grid;
  grid-template-columns : repeat(5, 1fr);
  align-items: center;
  justify-items: center;
  color: white;
`;

const NavItem = styled.li`
  list-style: none;
`;

const Logo = styled.img`
  height: 40px;
  width: auto;
`;

class Navigation extends React.Component {
  state = {};

  render() {
    return (
      <Nav>
        <UpperNav>
          <Logo src={require('./moviemania-logo.svg')}/>
        </UpperNav >
        <LowerNav>
          <NavItem>VIDEOS</NavItem>
          <NavItem>CHARACTERS</NavItem>
          <NavItem>COMICS</NavItem>
          <NavItem>MOVIES</NavItem>
          <NavItem>TV SHOWS</NavItem>
        </ LowerNav>
      </Nav>
    );
  }
}

export default Navigation;
