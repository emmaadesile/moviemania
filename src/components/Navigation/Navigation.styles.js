import styled from "styled-components";

const Nav = styled.nav`
  height: 92px;
  background: #22262a;
`;

const UpperNav = styled.div`
  height: 52px;
  border: 1px solid #393939;
  display: grid;
  grid-template-columns: 1fr;
  place-items: center;
`;

const LowerNav = styled.div`
  height: 40px;
  display: grid;
  grid-template-columns: 1fr;
  align-items: center;
  justify-items: center;
  color: white;

  .navItems {
    width: 200px;
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    justify-items: center;

    a {
      text-decoration: none;
      color: #fff !important;
      transition: 0.3s color ease-in-out;
      display: block;
    }

    a:hover {
      color: #777 !important;
    }

    li {
      list-style: none;
    }
  }
`;

const Logo = styled.img`
  height: 2.5rem;
  width: auto;
`;

export { Nav, UpperNav, LowerNav, Logo };
