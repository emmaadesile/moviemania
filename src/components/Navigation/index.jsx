import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import './moviemania-logo.svg';

const Nav = styled.nav`
  height: 92px;
  background: #22262A;
`;

const UpperNav = styled.div`
  height: 52px;
  border: 1px solid #393939;
  display: grid;
  grid-template-columns: 1fr;
  place-items: center;
`;

const LowerNav = styled.div`
  height: 40px;
  display: grid;
  grid-template-columns: 1fr;
  align-items: center;
  justify-items: center;
  color: white;

  .navItems {
    width: 200px;
    display: grid;
    grid-template-columns : repeat(2, 1fr);
    justify-items: center;

    a {
      text-decoration: none;
      color: #fff !important;
      transition: 0.3s color ease-in-out;
      display: block;
    }

    a:hover {
      color: #777 !important;
    }

    li {
      list-style: none;
    }
  }

`;

const Logo = styled.img`
  height: 2.5rem;
  width: auto;
`;

class Navigation extends React.Component {
  state = {};

  render() {
    return (
      <Nav>
        <UpperNav>
          <Logo src={require('../../assets/images/moviemania-logo.svg')}/>
        </UpperNav >
        <LowerNav>
          <div className='navItems'>
            <li><Link to="/movies">MOVIES</Link></li>
            <li><Link to="/tv">TV SHOWS</Link></li>
          </div>
        </LowerNav>
      </Nav>
    );
  }
}

export default Navigation;
