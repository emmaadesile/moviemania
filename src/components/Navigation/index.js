import React from "react";
import { Link } from "@reach/router";
import { Nav, UpperNav, LowerNav, Logo } from "./Navigation.styles";

class Navigation extends React.Component {
  state = {};

  render() {
    return (
      <Nav>
        <UpperNav>
          <Logo src={require("../../assets/images/moviemania-logo.svg")} />
        </UpperNav>
        <LowerNav>
          <div className="navItems">
            <li>
              <Link to="/movies">MOVIES</Link>
            </li>
            <li>
              <Link to="/tv">TV SHOWS</Link>
            </li>
          </div>
        </LowerNav>
      </Nav>
    );
  }
}

export default Navigation;
