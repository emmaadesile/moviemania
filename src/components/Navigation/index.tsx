import * as React from "react";
import { Link } from "@reach/router";
import { Nav, UpperNav, LowerNav, Logo } from "./Navigation.styles";
import LogoImg from "../../assets/images/moviemania-logo.svg";

const Navigation: React.FC = () => {
  return (
    <Nav>
      <UpperNav>
        <Link to="/">
          <Logo src={LogoImg} />
        </Link>
      </UpperNav>
      <LowerNav>
        <div className="navItems">
          <ul>
            <li>
              <Link to="/movies">MOVIES</Link>
            </li>
            <li>
              <Link to="/tv">TV SHOWS</Link>
            </li>
          </ul>
        </div>
      </LowerNav>
    </Nav>
  );
};

export default Navigation;
