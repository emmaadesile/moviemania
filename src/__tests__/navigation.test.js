import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Navigation from "../components/Navigation";

describe("App Navigation", () => {
  test('mounts', () => {
    const { container } = render(<Navigation />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("renders the navigation component", () => {
    const { getByText } = render(<Navigation />);
    const moviesLink = getByText(/movies/i);
    const tvShowLink = getByText(/tv shows/i);
    
    expect(moviesLink).toBeInTheDocument();
    expect(tvShowLink).toBeInTheDocument();
  });

  test('should redirect to tv-shows page', () => {
    const { getByText, debug } = render(<Navigation />)
    const tvShowsLink = getByText(/tv shows/i);

    fireEvent.click(tvShowsLink);
    debug()
  })
});
