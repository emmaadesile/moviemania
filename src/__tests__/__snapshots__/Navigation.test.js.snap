// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Navigation mounts 1`] = `
<nav
  class="sc-bdVaJa bEKecW"
>
  <div
    class="sc-bwzfXH gXdCbO"
  >
    <img
      class="sc-bxivhb cxjNSp"
      src="[object Object]"
    />
  </div>
  <div
    class="sc-htpNat bGvqhm"
  >
    <div
      class="navItems"
    >
      <li>
        <a
          href="/movies"
        >
          MOVIES
        </a>
      </li>
      <li>
        <a
          href="/tv"
        >
          TV SHOWS
        </a>
      </li>
    </div>
  </div>
</nav>
`;
